### PR TITLE
use coarser loci-ranges in TakeLociIterator

### PR DESCRIPTION
**Pull request by ryan-williams**
improves performance of capped-regions partitioner in cases where coverage is sparse; pathological cases would end up generating `LociSet`s with many 1000s of ranges comprising them.

### CODE REVIEW

File: src/main/scala/org/hammerlab/guacamole/loci/set/TakeLociIterator.scala [hunk 1]
```diff
@@ -97,8 +103,9 @@ class TakeLociIterator(it: BufferedIterator[(Position, Coverage)],
    * @return The number of covering regions, and a [[Contig]] with the taken loci ranges.
    */
   def takeLoci(it: ContigIterator[(HasLocus, Coverage)],
-            numRegions: Int,
-            dropAbove: Int): Option[(Int, Contig)] = {
+               numRegions: Int,
+               dropAbove: Int,
+               trimRanges: Boolean): Option[(Int, Contig)] = {
```
Comment (arahuja) [inline src/main/scala/org/hammerlab/guacamole/loci/set/TakeLociIterator.scala]: Add this to the doc above?

Comment (arahuja) [inline src/main/scala/org/hammerlab/guacamole/loci/set/TakeLociIterator.scala]: Add this to the doc above?

File: src/main/scala/org/hammerlab/guacamole/loci/set/TakeLociIterator.scala [hunk 2]
```diff
@@ -112,16 +119,16 @@ class TakeLociIterator(it: BufferedIterator[(Position, Coverage)],
...
+      if (intervals.isEmpty)
+        None
+      else
+        Some(
+          curNumRegions ->
+            Contig(
+              it.contigName,
+              intervals
+            )
+        )
```
Comment (arahuja) [inline src/main/scala/org/hammerlab/guacamole/loci/set/TakeLociIterator.scala]: Not sure what the diff is here? just spacing?

File: src/main/scala/org/hammerlab/guacamole/loci/set/TakeLociIterator.scala (hunk 1 above)
Comment (ryan-williams) [inline src/main/scala/org/hammerlab/guacamole/loci/set/TakeLociIterator.scala]: it was extraneous, this value comes in to the class via the constructor, I've removed it, good catch

File: src/main/scala/org/hammerlab/guacamole/loci/set/TakeLociIterator.scala (hunk 2 above)
Comment (ryan-williams) [inline src/main/scala/org/hammerlab/guacamole/loci/set/TakeLociIterator.scala]: yea this `if`/`else` was not indented from the `def` it implements before, and it should be

**Comment by ryan-williams** [comment]
> `trimRanges` is added by never set to true other than the tests - is that expected?

weird, I thought I'd added a cmdline-flag for it, but can't find it any of my branches, maybe I hadn't at all; anyway, I added it now, thanks for catching!

**Pull request merged**